# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender_build/src/shader_features.rs
# Commit: da24abf34123
# Full Hash: da24abf341236e00fa47e9e55dc72c82ae8a8c84
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-12-14 17:20:20
# Regressor Bug: 1866020
# File Overlap Count: 1
# Description:
#   Bug 1866020 - Override buggy colorspace conversion on Pixel devices. r=gw,padenot,geckoview-reviewers,owlish
#   
#   Pixel 6, 7, and 8 devices running Android 14 are affected by a bug
#   where video frames with SMPTE 432 color primaries are rendered
#   incorrectly when sampled from an external texture in GLES. To work
# ==============================================================================

diff -r 3a34e7a424e0 -r da24abf34123 gfx/wr/webrender_build/src/shader_features.rs
--- a/gfx/wr/webrender_build/src/shader_features.rs	Thu Dec 14 10:08:52 2023 +0000
+++ b/gfx/wr/webrender_build/src/shader_features.rs	Thu Dec 14 10:45:08 2023 +0000
@@ -15,7 +15,8 @@
         const DITHERING = 1 << 10;
         const TEXTURE_EXTERNAL = 1 << 11;
         const TEXTURE_EXTERNAL_ESSL1 = 1 << 12;
-        const DEBUG = 1 << 13;
+        const TEXTURE_EXTERNAL_BT709 = 1 << 13;
+        const DEBUG = 1 << 14;
     }
 }
 
@@ -138,6 +139,9 @@
     if flags.contains(ShaderFeatureFlags::TEXTURE_EXTERNAL) {
         texture_types.push("TEXTURE_EXTERNAL");
     }
+    if flags.contains(ShaderFeatureFlags::TEXTURE_EXTERNAL_BT709) {
+        texture_types.push("TEXTURE_EXTERNAL_BT709");
+    }
     let mut image_features: Vec<String> = Vec::new();
     for texture_type in &texture_types {
         let mut fast = FeatureList::new();